File: src/main/java/com/gmail/borlandlp/minigamesdtools/localizer/text/YAMLTextStorage.java
package com.gmail.borlandlp.minigamesdtools.localizer.text;

import com.gmail.borlandlp.minigamesdtools.Debug;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;

public class YAMLTextStorage implements LocalizerTextStorage {
    private YamlConfiguration storage;
    private File file;
    private String langCode;

    public YAMLTextStorage(File file) {
        this.storage = YamlConfiguration.loadConfiguration(file);
        this.file = file;
    }

    @Override
    public boolean containsText(String textId) {
        if(textId.contains(".")) {
            String msg = "textId[ID:" + textId + "] contains a directory delimiter for YAML. This may cause a malfunction of the plugin.";
            Debug.print(Debug.LEVEL.WARNING, msg);
        }

        return this.storage.contains(textId);
    }

    @Override
    public String getText(String textId) {
        if(textId.contains(".")) {
            String msg = "textId[ID:" + textId + "] contains a directory delimiter for YAML. This may cause a malfunction of the plugin.";
            Debug.print(Debug.LEVEL.WARNING, msg);
        }

        return this.storage.getString(textId);
    }

    @Override
    public String getLangCode() {
        return this.langCode;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/localizer/text/YAMLTextStorage.kt
package com.gmail.borlandlp.minigamesdtools.localizer.text

import com.gmail.borlandlp.minigamesdtools.Debug
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

class YAMLTextStorage(private val file: File) : LocalizerTextStorage {
    private val storage: YamlConfiguration = YamlConfiguration.loadConfiguration(file)
    override val langCode: String? = null

    override fun containsText(textId: String): Boolean {
        if (textId.contains(".")) {
            val msg =
                "textId[ID:$textId] contains a directory delimiter for YAML. This may cause a malfunction of the plugin."
            Debug.print(
                Debug.LEVEL.WARNING,
                msg
            )
        }
        return storage.contains(textId)
    }

    override fun getText(textId: String): String? {
        if (textId.contains(".")) {
            val msg =
                "textId[ID:$textId] contains a directory delimiter for YAML. This may cause a malfunction of the plugin."
            Debug.print(
                Debug.LEVEL.WARNING,
                msg
            )
        }
        return storage.getString(textId)
    }

}